CONTRIBUTING.md

```diff
@@ -47,3 +47,73 @@ mvn -f sdk/{projectForlderDir}/pom.xml -Dgpg.skip clean install
 
 //example: mvn -f sdk/keyvault/azure-keyvault-keys/pom.xml clean install
 ```
+
+## Versions and versioning
+Tooling has been introduced to centralize versioning and help ease the pain of updating artifact versions in POM and README files. Under the eng\versioning directory there version text files, one for client (version_client.txt) and one for data (version_data.txt)*. The format of the version files is as follows:
+groupId:artifactId;dependency-version;current-version
+The dependency-version should be set to the most recent released version and the current-version is set to the next version to be released. 
+For example:
+com.azure:azure-identity;1.0.0-preview.4;1.0.0-preview.5
+Note: In the case of a new or unreleased artifact both versions will be the same.
+
+### Libraries vs External Dependencies
+Libraries refer to things that are built and released as part of the Azure SDK. Libraries have a current version and a dependency version. 
+
+External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have one version.
+
+### Current version vs Depdendency version
+Current version - This is the version we should be using when defining a component in its POM file and also when dependent components are built within the same pipeline. The current version is the version currently in development.
+Depdendency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package.
+
+For example: com.azure:azure-storage-blob-batch has dependencies on com.azure:azure-core, com.azure:azure-core-http-netty and com.azure:azure-storage-blob. Because com.azure:azure-core and com.azure:azure-core-http-netty are both built outside of azure-storage pipeline we should be using the released or Dependency versions of these when they're dependencies of another library. Similarly, libraries built as part of the same pipeline, that have interdependencies, should be using the Current version. Since com.azure:azure-storage-blob-batch and com.azure:azure-storage-blob are both built part of the azure-batch pipeline when com.azure:azure-storage-blob is declared as a dependency of com.azure:azure-storage-blob-batch it should be the Current version.
```

Comment:
I suggest wrapping these artifact names with a backtick so that they stand out when reading in md.

Reply:
Updated. Next commit will have the change.

Reply:
Updated. Will be updated with the next commit.